syntax = "proto3";

package app.api_error_model;

import "google/protobuf/any.proto";

enum ErrorType {
    INTERNAL = 0;
    EXTERNAL_SERVICE = 1;
    VALIDATION = 2;
    NOT_USER_FAULT = 3;
}

message ErrorResponse {
    ErrorType error_type = 1;
    string error_code = 2;
    string message = 3;
    ErrorResponse caused_by = 4;
    google.protobuf.Any data = 5;
}